http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppComponent.scala

```diff
@@ -239,6 +239,11 @@ object appQuery extends TableQuery(new AppTable(_)) {
       .map(a => (a.status, a.destroyedDate, a.diskId))
       .update((AppStatus.Deleted, now, None))
 
+  def updateDateAccessed(appName: AppName, now: Instant): DBIO[Int] =
```

Comment:
looks like starting with the nonDeleted filter is what's missing the index - `destroyedDate` is in the indexes but only 2nd or 3rd down the list, and `status` isn't at all I think. if we started with `appName` we'd be better off

Reply:
yea, I was a bit suspicious when looking at this..but didn't feel like fixing the existing problem....will take another look

Reply:
so I think these are the only indexes
```
  PRIMARY KEY (`id`),
  UNIQUE KEY `IDX_APP_UNIQUE` (`appName`,`nodepoolId`,`workspaceId`,`destroyedDate`),
```

not sure what would be good index to add..the real meaningful uniqueness would be (`appName`, `workspaceId`) or `appName`, `cloudContext`)...

Anyways, it's probably not an issue today since we only have 3385 records in prod DB...we don't update `dateAccessed` for every proxy request either, we batch update every 30 seconds...

Another good motivation to rethink DB schema.